### ISSUE TITLE
Cannot use Cython summarize_data when data are bad in a certain way

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#44]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/44) by Jim_HW (Bitbucket: [Jim_HW](https://bitbucket.org/Jim_HW), ).**

----------------------------------------

Joe and Jim here together, cannot run TESGroup.summarize_data on a dataset where there are lots of unlock events that don't always recover before the next record.

We surmise that the following is the problem pulse.... or *one* problem pulse.

The actual error was VERY hard to identify, because the channel_group.py function ```summarize_data``` catches and hides all errors. (Consider this another big problem.) But fortunately the ```MicrocalDataSet.summarize_data``` provides the following traceback:

```

#!python


In [6]: ds.summarize_data(60, use_cython=False)
---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
<ipython-input-6-df9196a28919> in <module>()
----> 1 ds.summarize_data(60, use_cython=False)

mass/core/cython_channel.pyx in mass.core.cython_channel.CythonMicrocalDataSet.summarize_data (mass/core/cython_channel.c:2288)()

mass/core/channel.pyx in mass.core.channel.MicrocalDataSet.summarize_data (mass/core/channel.c:25776)()

ZeroDivisionError: float division


```

Joe is pretty sure this error occurs in 

```
#!python

                p_pretrig_mean_array[j] = <float>ptm
                p_pretrig_rms_array[j] = <float>ptrms
                peak_value -= <unsigned short>ptm
                p_promptness_array[j] = (promptness_sum / 6.0 - ptm) / peak_value
                p_peak_value_array[j] = peak_value
                p_peak_index_array[j] = peak_index
                p_min_value_array[j] = min_value

```
when peak_value is computed to be zero (presumably it was *less than zero* and then truncated to zero).
![example_uncrunchable_data.png](https://bitbucket.org/repo/KrBpjo/images/3468457967-example_uncrunchable_data.png)

**Comment by joefowler** [comment]
**Original comment by Young Il Joe (Bitbucket: [jojo978](https://bitbucket.org/jojo978), GitHub: [jojo978](https://github.com/jojo978)).**

----------------------------------------

Hi Jim,

I think you have still lingering old channel.pyx. I think you can remove the installed mass (somewhere under ?/site-packages/mass) manually (by rm -rf mass). With use_cython=False, you can use pure python summarize_data routine.

For these data (peak_value < ptm), what do we want to set p_peak_vallue, p_promptness, and p_peak_index to?

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Sorry Young. We copied the text from channel.pyx THEN installed branch develop. That branch had the same problem, but the error message you way is different (of course).

I have a proposal, which I will test and push to develop soon.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Properly handle a case where the pretrigger mean exceeds the highest post-trig value.

Before, that created a divide-by-zero error in some cases. Now the nonsense values
of 0 are stored for the promptness, the peak value, and the peak index.

Fixes #44 (I think, though that might be a premature statement....)
